patches/net-socket-socks5_client_socket.cc.patch

```diff
@@ -12,42 +12,32 @@ index 575c1dcd1bc31047ff0155f3287ac9dc145307f1..645139d09cef2744d384ddf05daef964
        case STATE_HANDSHAKE_WRITE:
          DCHECK_EQ(OK, rv);
          net_log_.BeginEvent(NetLogEventType::SOCKS5_HANDSHAKE_WRITE);
-@@ -252,8 +255,6 @@ int SOCKS5ClientSocket::DoLoop(int last_io_result) {
-   return rv;
- }
- 
--const char kSOCKS5GreetWriteData[] = { 0x05, 0x01, 0x00 };  // no authentication
--
- int SOCKS5ClientSocket::DoGreetWrite() {
-   // Since we only have 1 byte to send the hostname length in, if the
-   // URL has a hostname longer than 255 characters we can't send it.
-@@ -263,8 +264,12 @@ int SOCKS5ClientSocket::DoGreetWrite() {
+@@ -263,8 +266,11 @@ int SOCKS5ClientSocket::DoGreetWrite() {
    }
  
-   if (buffer_.empty()) {
--    buffer_ =
--        std::string(kSOCKS5GreetWriteData, std::size(kSOCKS5GreetWriteData));
-+    const char greeting[] = {
-+      0x05,			// SOCKS version
-+      0x01,			// number of authentication methods
-+      static_cast<char>(auth_method()),
-+    };
-+    buffer_ = std::string(greeting, sizeof(greeting));
-     bytes_sent_ = 0;
+   if (!write_buf_) {
+-    auto greet_buffer =
+-        base::MakeRefCounted<WrappedIOBuffer>(kSOCKS5GreetWriteData);
++    static_assert(kSOCKS5GreetWriteData.size() == 3u);
```

Comment:
this is a very old patch, we should update to latest patching standards to clean this up

Reply:
I wonder if we even need this anymore? I can't remember why it was added

Reply:
@diracdeltas do you remember what this was for? Maybe it's something we can upstream?

Reply:
Standardized the patch in https://github.com/brave/brave-core/pull/27522/commits/edc7cc7a70ca80e3a6ea4254a3b4311476be6fdb.